public/scripts/project/lock.js

```diff
@@ -0,0 +1,75 @@
+define(function(require) {
+  var fs = Bramble.getFileSystem();
+  var sh = new fs.Shell();
+  var LOCK_KEY = 'bramble-lock';
+  var EXPIRY = 2000; // MS
+
+  function lock(root, callback) {
+    function refresh(callback) {
+      fs.setxattr(root, LOCK_KEY, Date.now(), function(err) {
+        if (err) {
+          return callback(err);
+        }
+
+        callback();
+      });
+    }
+
+    function setLock() {
+      function handleError(err) {
+        console.error("[Bramble] Failed to refresh project lock");
+      }
+
+      refresh(function(err) {
+        if (err) {
+          return callback(err);
+        }
+
+        setInterval(refresh.bind(null, handleError), EXPIRY);
+        callback();
+      });
+    }
+
+    isLocked(root, function(err) {
+      if (err) {
+        return callback(err);
+      }
+
+      setLock();
+    });
+  }
+
+  function isLocked(root, callback) {
+    // First, make sure the directory exists
+    sh.mkdirp(root, function(err) {
+      if (err && err !== "EEXIST") {
+        return callback(err);
+      }
+
+      // Then we try and access the lock
+      fs.getxattr(root, LOCK_KEY, function(err, value) {
+        if (err) {
+          if (err !== "ENOENT") {
```

Comment:
Do you also explicitly want to deal with the attribute not existing?  Filer will return that error if it doesn't.